src/pymor/models/neural_network.py

```diff
@@ -202,6 +202,83 @@ def _compute_solution(self, mu=None, **kwargs):
+
+        Parameters
+        ----------
+        T
+            The final time T.
+        nt
+            The number of time steps.
+        neural_network
+            The neural network that approximates the mapping from parameter space
+            to solution space. Should be an instance of
+            :class:`~pymor.models.neural_network.FullyConnectedNN` with input size that
+            matches the (total) number of parameters and output size equal to the
+            dimension of the reduced space.
+        parameters
+            |Parameters| of the reduced order model (the same as used in the full-order
+            model).
+        output_functional
+            |Operator| mapping a given solution to the model output. In many applications,
+            this will be a |Functional|, i.e. an |Operator| mapping to scalars.
+            This is not required, however.
+        products
+            A dict of inner product |Operators| defined on the discrete space the
+            problem is posed on. For each product with key `'x'` a corresponding
+            attribute `x_product`, as well as a norm method `x_norm` is added to
+            the model.
+        error_estimator
+            An error estimator for the problem. This can be any object with
+            an `estimate_error(U, mu, m)` method. If `error_estimator` is
+            not `None`, an `estimate_error(U, mu)` method is added to the
+            model which will call `error_estimator.estimate_error(U, mu, self)`.
+        visualizer
+            A visualizer for the problem. This can be any object with
+            a `visualize(U, m, ...)` method. If `visualizer`
+            is not `None`, a `visualize(U, *args, **kwargs)` method is added
+            to the model which forwards its arguments to the
+            visualizer's `visualize` method.
+        name
+            Name of the model.
+        """
+
+        def __init__(self, T, nt, neural_network, parameters={}, output_functional=None,
+                     products=None, error_estimator=None, visualizer=None, name=None):
+
+            super().__init__(products=products, error_estimator=error_estimator,
+                             visualizer=visualizer, name=name)
+
+            self.__auto_init(locals())
+            self.solution_space = NumpyVectorSpace(neural_network.output_dimension)
+            if output_functional is not None:
+                self.dim_output = output_functional.range.dim
+
+        def _compute_solution(self, mu=None, **kwargs):
+
+            U = self.solution_space.empty(reserve=self.nt)
+
+            # create tensor of parameters at different time instances
+            def time_dependent_parameter(t):
+                return [mu.get_time_dependent_value(param)(t) for param in mu]
+
+            parameters = torch.DoubleTensor([[time_dependent_parameter(t)
+                                              for t in np.linspace(0., self.T, self.nt)]])
```

Comment:
Did I get it right that this should actually be something like the following?
```suggestion
            parameters = torch.DoubleTensor([[mu.with_(t=t)
                                              for t in np.linspace(0., self.T, self.nt)]])
```

Reply:
Yes, I think so.

Reply:
The `to_numpy()` seems to be missing, though.

Reply:
Also, if I play with the expression locally, I get

```
UserWarning: Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with numpy.array() before converting to a tensor.
```

Reply:
If I use `mu.with_(t=t).to_numpy()`, the result also includes the time-component, which I do not want to pass to the network in this case. So this does not really fit the use case here. For the moment it could also be fine to just restrict our attention to parameters that do not depend on time. What do you think @sdrave?

Reply:
Right, I think your solution is probably the best approach. Be aware though, that the `mu` passed to solve could already contain a `t` or other parameters which have nothing to do with the model. Maybe not very likely but not illegal. So it would probably better to write:

```python
def time_dependent_parameter(t):
    return [mu.get_time_dependent_value(param)(t) for param in self.parameters]
```